packages/toolpad-components/src/Form.tsx

```diff
@@ -172,24 +170,27 @@ export default createComponent(Form, {
   },
 });
 
-export interface FormInputComponentProps {
+export interface FormInputComponentProps extends FormInputValidationProps {
   name: string;
+}
+
+export interface FormInputValidationProps {
   isRequired: boolean;
   minLength: number;
   maxLength: number;
   isInvalid: boolean;
 }
 
+type ValidationProps = Partial<FormInputValidationProps>;
```

Comment:
I would not separate this type into `ValidationProps` as it is not reused anywhere - I would just write ```validationProps: Partial<FormInputValidationProps>``` below. 
But this is not a big deal!